lib/ruby_smb/dcerpc/icpr.rb

```diff
@@ -72,7 +72,7 @@ def cert_server_request(attributes:, authority:, csr:)
               status_code: status_code
             )
           end
-        else
```

Comment:
Super minor, but based on this and the definition for `ret` on L, `ret` will not contain the key `certificate`.  I'm assuming that's expected, though typically it makes me feel warmer and fuzzier to have `ret[:certificate]` be `nil` rather than undefined.  I don't know if we want to add the `certificate` key to the original `ret` definition, or if we always check or catch the key exception when we try to access it.

Reply:
Yeah the `:certificate` key won't be set when the status is error, so callers should already account for certificate only being present sometimes. I can get it updated so `:certificate` is initialized in `ret` to be nil though. 👍🏻